decoder.go

```diff
@@ -138,12 +137,17 @@ func NewDecoder(r io.Reader) *Decoder {
 }
 
 func (dec *Decoder) readLine() ([]byte, error) {
-	b, err := dec.br.ReadSlice('\n')
-	b = bytes.TrimRight(b, "\r\n")
-	if err == io.EOF && len(b) > 0 {
-		err = nil
+	line, isPrefix, err := dec.br.ReadLine()
+	if isPrefix {
```

Comment:
To avoid repeating the logic twice, one could do something like

```go
var buf []byte
for {
    line, isPrefix, err := dec.br.ReadLine()
    if !isPrefix && len(buf) == 0 {
        return line, err
    }

    buf = append(buf, line...)
    if !isPrefix {
        return buf, err
    }
}
```

Apart from that, LGTM!

Reply:
Let me know if the latest version looks good. Thanks for the feedback!